ui/src/ducks/app/solutions.js

```diff
@@ -0,0 +1,176 @@
+import { call, put, takeEvery, select, delay } from 'redux-saga/effects';
+import * as ApiK8s from '../../services/k8s/api';
+import history from '../../history';
+import { REFRESH_TIMEOUT } from '../../constants';
+
+const APP_K8S_PART_OF_SOLUTION_LABEL = 'app.kubernetes.io/part-of';
+const APP_K8S_VERSION_LABEL = 'app.kubernetes.io/version';
+
+// Actions
+export const SET_SOLUTIONS = 'SET_SOLUTIONS';
+export const SET_SOLUTIONS_REFRESHING = 'SET_SOLUTIONS_REFRESHING';
+export const SET_SERVICES = 'SET_SERVICES';
+export const SET_ENVIRONMENTS = 'SET_ENVIRONMENTS';
+const CREATE_ENVIRONMENT = 'CREATE_ENVIRONMENT';
+const REFRESH_SOLUTIONS = 'REFRESH_SOLUTIONS';
+const STOP_REFRESH_SOLUTIONS = 'STOP_REFRESH_SOLUTIONS';
+
+// Reducer
+const defaultState = {
+  solutions: [],
+  services: [],
+  environments: [],
+  isSolutionsRefreshing: false,
+};
+
+export default function reducer(state = defaultState, action = {}) {
+  switch (action.type) {
+    case SET_SOLUTIONS:
+      return { ...state, solutions: action.payload };
+    case SET_SERVICES:
+      return { ...state, services: action.payload };
+    case SET_ENVIRONMENTS:
+      return { ...state, environments: action.payload };
+    case SET_SOLUTIONS_REFRESHING:
+      return { ...state, isSolutionsRefreshing: action.payload };
+    default:
+      return state;
+  }
+}
+
+// Actions Creator
+
+export function setSolutionsAction(solutions) {
+  return { type: SET_SOLUTIONS, payload: solutions };
+}
+
+export function setSolutionsRefeshingAction(payload) {
+  return { type: SET_SOLUTIONS_REFRESHING, payload };
+}
+
+export function setServicesAction(services) {
+  return { type: SET_SERVICES, payload: services };
+}
+
+export const setEnvironmentsAction = environments => {
+  return { type: SET_ENVIRONMENTS, payload: environments };
+};
+
+export const refreshSolutionsAction = () => {
+  return { type: REFRESH_SOLUTIONS };
+};
+
+export const stopRefreshSolutionsAction = () => {
+  return { type: STOP_REFRESH_SOLUTIONS };
+};
+
+export function createEnvironmentAction(newEnvironment) {
+  return { type: CREATE_ENVIRONMENT, payload: newEnvironment };
+}
+
+export function* createEnvironment(action) {
+  const newEnvironment = action.payload;
+
+  const body = {
+    apiVersion: 'solutions.metalk8s.scality.com/v1alpha1',
+    kind: 'Environment',
+    metadata: {
+      name: newEnvironment.name,
+    },
+    spec: {
+      description: newEnvironment.description,
+      solutions: [],
+    },
+  };
+
+  const result = yield call(ApiK8s.createEnvironment, body);
+  if (!result.error) {
+    yield call(fetchEnvironments);
+    history.push('/solutions');
+  }
+  return result;
+}
+
+export function* fetchUIServices() {
+  const result = yield call(ApiK8s.getUIServiceForAllNamespaces);
+  if (!result.error) {
+    yield put(setServicesAction(result.body.items));
+  }
+  return result;
+}
+
+export function* fetchSolutions() {
```

Comment:
This is a complexe saga. let's a unit tests but can be done later

Reply:
Issue created #1813